src/core/execute.c

```diff
@@ -3510,7 +3519,18 @@ static int exec_child(
          * This needs to be done after PrivateDevices=y setup as device nodes should be owned by the host's root.
          * For non-root in a userns, devices will be owned by the user/group before the group change, and nobody. */
         if (needs_setuid) {
-                r = enforce_groups(gid, supplementary_gids, ngids);
+                _cleanup_free_ gid_t *gids_to_enforce = NULL;
+                int ngids_to_enforce = 0;
+
+                ngids_to_enforce = merge_gid_lists(supplementary_gids, ngids, gids_after_pam, ngids_after_pam, &gids_to_enforce);
+
```

Comment:
please remove empty line, we generally want to keep function calls and their error handling together.

also please line break this long line